ui/src/app/components/app-router.tsx

```diff
@@ -59,3 +59,16 @@ export const Navigate = ({to}): React.ReactElement => {
   const location = useLocation();
   return <Redirect to={{pathname: to, state: {from: location}}}/>;
 };
+
+// Resets the state after change to prevent a navigation loop
```

Comment:
It would help to  have a little more documentation for what this does and doesn't do for you.  Maybe name it  `useGatedNavigation`?

Reply:
Changed to `useNavigationState` and updated the comment